copyright/copyright.js

```diff
@@ -0,0 +1,106 @@
+/* * *  *  * *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * */
+/* Copyright (c) 2017 Mobify Research & Development Inc. All rights reserved. */
+/* * *  *  * *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * */
+
+const glob = require('glob')
+const fs = require('fs')
+const path = require('path')
+
+const red = '\x1b[31m'
+const green = '\x1b[32m'
+const yellow = '\x1b[33m'
+const magenta = '\x1b[35m'
+const cyan = '\x1b[36m'
+const blackBG = '\x1b[40m'
+const defaultBG = '\x1b[49m'
+const defaultFG = '\x1b[39m'
+
+const args = process.argv.filter((arg) => {
+    return !/node|copyright/.test(arg)
+})
+
+const copyright = {
+    lintMode: true,
+    langs: {},
+    run() {
+        this.buildSupportedExtensions()
+        let files = []
+
+        const processedGlobs = args.map((folder) => {
+            files.push(glob.sync(folder))
+        })
+
+        // flattens array of arrays returned by glob.sync
+        files = files.reduce((a, b) => a.concat(b))
+
+        const filesContainingHeader = files.filter((file) => {
+            const content = fs.readFileSync(file)
+            const hasCopyrightHeader = content.includes('Copyright (c)')
+            const ext = file.match(/\.[0-9a-z]+$/i)[0]
+
+            if (hasCopyrightHeader) {
+                return true
+            } else {
+                if (this.lintMode) {
+                    console.log(`${yellow}${file} ${red}missing copyright header`)
+                    return false
+                } else {
+                    const newData = this.getHeaderText(ext) + content
+                    fs.writeFileSync(file, newData)
+                    console.log(`${green}Copyright header succesfully written into ${magenta}${file}`)
+                    return true
+                }
+            }
+        })
```

Comment:
Thinking about this some more, we don't really need this array, just a flag for whether an error has occurred. Then this can be a `forEach` that just sets the flag rather than worrying about return values.